libparsec/crates/platform_mountpoint/src/unix/mod.rs

```diff
@@ -0,0 +1,380 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+mod entry_info;
+mod inode;
+
+use fuser::{
+    FileType, Filesystem, MountOption, ReplyAttr, ReplyCreate, ReplyData, ReplyDirectory,
+    ReplyEmpty, ReplyEntry, ReplyOpen, ReplyWrite, Request, Session, SessionUnmounter,
+};
+use libc::{ENAMETOOLONG, ENOENT, ENOTEMPTY};
+use libparsec_types::{anyhow, FileDescriptor};
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use crate::{error::MountpointError, MountpointInterface, MountpointManager, WriteMode};
+
+use inode::Inode;
+pub(crate) use inode::InodeManager;
+
+/// TODO: Do we need to change this value ?
+/// Validity timeout
+const TTL: Duration = Duration::ZERO;
+
+/// TODO: Do we need to handle any other GENERATION ?
+///
+/// If the file system will be exported over NFS, the ino/generation pairs need
+/// to be unique over the file system's lifetime (rather than just the mount
+/// time). So if the file system reuses an inode after it has been deleted, it
+/// must assign a new, previously unused generation number to the inode at the
+/// same time.
+const GENERATION: u64 = 0;
+
+impl Filesystem for MountpointManager {
```

Comment:
We don't want to have this code being aware of the actual implementations of filesystem (otherwise we will have to do one implement per type of filesystem)

Instead you should do `impl<T: MountpointInterface> FileSystem for T { ... }`

Reply:
It is possible only if `FileSystem` trait is in the current crate

Reply:
```Shell
error[E0210]: type parameter `T` must be used as the type parameter for some local type (e.g., `MyStruct<T>`)
  --> libparsec/crates/platform_mountpoint/src/unix/mod.rs:36:6
   |
36 | impl<T: MountpointInterface> Filesystem for T {
   |      ^ type parameter `T` must be used as the type parameter for some local type
   |
   = note: implementing a foreign trait is only possible if at least one of the types for which it is implemented is local
   = note: only traits defined in the current crate can be implemented for a type parameter

For more information about this error, try `rustc --explain E0210`.
error: could not compile `libparsec_platform_mountpoint` (lib) due to previous error
```